packages/core/src/bind/connectFactoryObservable.ts

```diff
@@ -28,30 +74,30 @@ import { takeUntilComplete } from "../internal/take-until-complete"
  * subscription, then the hook will leverage React Suspense while it's waiting
  * for the first value.
  */
-export default function connectFactoryObservable<
-  A extends (number | string | boolean | null)[],
-  O
->(
+export default function connectFactoryObservable<A extends [], O>(
   getObservable: (...args: A) => Observable<O>,
   unsubscribeGraceTime: number,
 ): [
   (...args: A) => Exclude<O, typeof SUSPENSE>,
   (...args: A) => Observable<O>,
 ] {
-  const cache = new Map<string, [Observable<O>, BehaviorObservable<O>]>()
+  const cache = new NestedMap<A, [Observable<O>, BehaviorObservable<O>]>()
 
   const getSharedObservables$ = (
-    ...input: A
+    input: A,
   ): [Observable<O>, BehaviorObservable<O>] => {
-    const key = JSON.stringify(input)
-    const cachedVal = cache.get(key)
+    for (let i = input.length - 1; input[i] === undefined && i > -1; i--) {
+      input.splice(-1)
+    }
+    const keys = ([input.length, ...input] as any) as A
```

Comment:
This reminded me of something exciting: The TS release for this month (4.0) will include [variadic tuple types](https://github.com/microsoft/TypeScript/pull/39094) - Which will allow us to represent this type as `[number, ...A]`.